### ISSUE TITLE
abi.encodeCall not works with interfaces

### ISSUE THREAD

**Issue by k06a** [ethereum/solidity#12448]
## Description

I expect `abi.encodeCall` to work with function pointers from both contracts and interfaces. But now it fails with strange error when function pointer comes from interface like this:

```solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.11;

interface IERC20 {
    function balanceOf(address account) external view returns(uint256);
    function transfer(address to, uint256 amount) external returns(bool);
    function transferFrom(address spender, address to, uint256 amount) external returns(bool);
}

contract Foo {
    function bar(IERC20 token, uint256 amount) external view returns(bytes memory) {
        return abi.encodeCall(
            IERC20.transferFrom,
            (
                msg.sender,
                address(this),
                amount
            )
        );
    }
}
```

```
TypeError: Function must be "public" or "external".
  --> contracts/1_Storage.sol:14:13:
   |
14 |             IERC20.transferFrom,
   |             ^^^^^^^^^^^^^^^^^^^
Note: Function is declared here:
 --> contracts/1_Storage.sol:8:5:
  |
8 |     function transferFrom(address spender, address to, uint256 amount) external returns(bool);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Changing `IERC20.tranferFrom` to `IERC20(address(0)).transferFrom` works, but I'm sure this should be considered as bug and fixed.

## Environment

- Compiler version: 0.8.11
- Target EVM version (as per compiler settings): default

**Comment by k06a** [comment]
Duplicate of: https://github.com/ethereum/solidity/issues/12435

**Comment by cameel** [comment]
Yeah, this will be fixed in #12437.
